tests/backend/test_app_session_save.py

```
@@ -0,0 +1,97 @@
+def test_post_save_side_effect_failure_returns_success(
+    monkeypatch: pytest.MonkeyPatch, tmp_path: Path
+) -> None:
+    """Report success once persistence committed even if later work fails."""
+    calls: list[str] = []
+    monkeypatch.setattr(app, "load_session", lambda _label: {})
+    monkeypatch.setattr(app, "get_session_path", lambda _label: tmp_path / "session.yaml")
+    monkeypatch.setattr(app, "save_session", lambda *_a, **_k: calls.append("save"))
+
+    def fail_clear(_label: str) -> None:
+        calls.append("clear")
+        raise RuntimeError("event log unavailable")
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)